src/main/java/com/deviceinsight/kafka/health/KafkaConsumingHealthIndicator.java

```diff
@@ -134,27 +144,15 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
 
 	private KafkaCommunicationResult sendAndReceiveMessage() {
 
-		Future<Void> sendReceiveTask = null;
-
 		try {
-
-			sendReceiveTask = executor.submit(() -> {
-				sendAndReceiveKafkaMessage();
-				return null;
-			});
-
-			sendReceiveTask.get(sendReceiveTimeoutMs, MILLISECONDS);
-
-		} catch (ExecutionException e) {
-			logger.warn("Kafka health check execution failed.", e);
-			return KafkaCommunicationResult.failure(topic, e);
-		} catch (TimeoutException | InterruptedException e) {
-			logger.warn("Kafka health check timed out.", e);
-			sendReceiveTask.cancel(true);
-			return KafkaCommunicationResult.failure(topic, e);
+			sendAndReceiveKafkaMessage();
 		} catch (RejectedExecutionException e) {
 			logger.debug("Ignore health check, already running...");
+		} catch (Exception e) {
```

Comment:
It's not clear, it depends on the call. If the producer fail it will be InterruptedException, ExecutionException, TimeoutException, but the consumer throw more kafka related exceptions.

I think it'll be best to list the exceptions from before.